skimage/io/tests/test_collection.py

```diff
@@ -104,29 +114,25 @@ def set_files(f):
             set_files('newfiles')
 
     @pytest.mark.skipif(not has_pooch, reason="needs pooch to download data")
-    def test_custom_load_func_sequence(self):
-        filename = fetch('data/no_time_for_that_tiny.gif')
-
+    def test_custom_load_func_sequence(self, random_gif_path):
         def reader(frameno):
-            vid = imageio.get_reader(filename)
-            return vid.get_data(frameno)
```

Comment:
@FirefoxMetzger, I couldn't figure out an equivalent version of `get_reader` and `get_data` with the v3 API. It seems like `get_reader` will be deprecated? `imageio.v3.imiter` comes close but its returned iterable is not index-able.

Reply:
Ah, nevermind. I think 
```python
file = imageio.v3.imopen(filename, "r")
file.read(index=0)
```
should be equivalent.

Reply:
> I couldn't figure out an equivalent version of get_reader and get_data with the v3 API

That depends a bit on what data you want. For the same level of generality `imopen` is what you want. It's actually even more general than `get_reader`. Here, it's probably `imread` combined with the `index=` kwarg.

The reason we move away from `get_reader` is because `imopen`, `imread`, and `imiter` combined cover all the relevant use cases. If you want a single ndimage use `imread` (potentially with `index=`), if you want to iterate over all ndimages use `imiter`, and if you try to minimize re-opening the file or need some plugin-specific logic use `imopen`,

It's also one of the most misused functions in ImageIO. `file = get_reader(...)` is as "bad" as `file = open(...)` since most users don't `file.close()` it explicitly. As a result, they rely on implementation details of CPython's GC to close the file, which is a _very_ bad idea for long-running tasks or notebooks that will just keep the file open forever. It also breaks on other Python implementations, say pypy, because Python's standard doesn't require a GC to ref-count (and pypy explicitly chooses not to for speed). 

`imread` and `imiter` have a clear scope on when to open/close a file and will thus clean up after themselves. `imopen` has the same "problem" as `open` or `get_reader` and one should always use it with a context manager to ensure cleanup. Online code snippets (which inevitably end up being copy-pasted) for `open` and `imopen` do, but those for `get_reader` don't; hence the antipattern proliferates. `get_reader` has supported a context manager since forever, Unfortunately, nobody seems interested in using it, which is one more reason to nudge people away from `get_reader`.